tap_github/client.py

```diff
@@ -266,10 +271,86 @@ def parse_response(self, response: requests.Response) -> Iterable[dict]:
         resp_json = response.json()
         yield from extract_jsonpath(self.query_jsonpath, input=resp_json)
 
+    def get_next_page_token(
+        self, response: requests.Response, previous_token: Optional[Any]
+    ) -> Optional[Any]:
+        """
+        Return a dict of cursors for identifying next page or None if no more pages.
+
+        Note - pagination requires the Graphql query to have nextPageCursor_X parameters
+        with the assosciated hasNextPage_X, startCursor_X and endCursor_X.
+
+        X should be an integer between 0 and 9, increasing with query depth.
+        """
+
+        resp_json = response.json()
+        # Find if results contains "hasNextPage_X" flags and if any are True.
+        # If so, set nextPageCursor_X to endCursor_X for X max.
+
+        next_page_results = collections.OrderedDict(
+            nested_lookup(
+                key="hasNextPage",
+                document=resp_json,
+                wild=True,
+                with_keys=True,
+            )
+        )
+
+        has_next_page_results = collections.OrderedDict()
+        # Iterate over all the items and filter items with hasNextPage = True.
+        for (key, value) in next_page_results.items():
+            # Check if key is even then add pair to new dictionary
+            if any(value):
+                has_next_page_results[key] = value
+
+        # Check if any "hasNextPage" is True.
+        resp_has_next_page = bool(has_next_page_results)
+
+        # No pagination necessary, exit early.
+        if not resp_has_next_page:
+            return None
+
+        # Get deepest pagination item
+        (pagination_key, _) = next(reversed(has_next_page_results.items()))
```

Comment:
I don't know what the results look like, but is it guaranteed that the pagination keys will be ordered?